### ISSUE TITLE
Updating git

### ISSUE THREAD

**Issue by p3lim** [babun/babun#232]
So I just got this, first thing I noticed is the old version of git that was shipped with it (1.7.9) and my first task was to update this.

I found no way of updating it in pact, so I tried removing it then adding it again.
It found the version 2.1.4 and installed that, but now git doesn't work.

**Comment by RMAdevelopper** [comment]
Same here.

The command doens't prompt any errors, output just remains empty.
Same behaviour with `babun check`, since the git update...

**Comment by tombujok** [comment]
So you did git uninstall and then git install (both with pact) and after
that git does not work?

**Comment by RMAdevelopper** [comment]
Exactly. 
I did it again, here is the whole install output  with the 2 git commands at the end:

```bash
{ ~ }  » pact install git                                                                   ~ 127
Working directory is /setup
Mirror is http://mirrors.kernel.org/sourceware/cygwin/
setup.ini taken from the cache

Installing git
Found package git
File ‘git-2.1.4-1.tar.xz’ already there; not retrieving.

Unpacking...
Package git requires the following packages, installing:
bash libcurl4 libexpat1 libgcc1 libiconv2 libintl8 libopenssl100 perl perl-Error perl_vendor python zlib0 cygutils less openssh rsync _update-info-dir cygwin
Package bash is already installed, skipping
Package libcurl4 is already installed, skipping
Package libexpat1 is already installed, skipping
Package libgcc1 is already installed, skipping
Package libiconv2 is already installed, skipping
Package libintl8 is already installed, skipping
Package libopenssl100 is already installed, skipping
Package perl is already installed, skipping
Package perl-Error is already installed, skipping
Package perl_vendor is already installed, skipping
Package python is already installed, skipping
Package zlib0 is already installed, skipping
Package cygutils is already installed, skipping
Package less is already installed, skipping
Package openssh is already installed, skipping
Package rsync is already installed, skipping
Package _update-info-dir is already installed, skipping
Package cygwin is already installed, skipping
Package git installed
{ ~ }  » git --version                                                                          ~
{ ~ }  » git status                                                                         ~ 127
{ ~ }  »          
```

**Comment by p3lim** [comment]
@tombujok That's exactly what I did, as I found no other means of updating. Now git doesn't seem to work (I can execute commands, but nothing it output, not even errors).

**Comment by v-yadli** [comment]
Same here. It's because the new version of (almost anything in the repo actually) git rely on a newer version of libgcc1. However we cannot simply update libgcc1. It's about the whole environment.

**Comment by v-yadli** [comment]
@tombujok Is there a way to update the whole base system?

**Comment by tombujok** [comment]
I understand the problem now. Yes, definitely there is - babun is a normal cygwin installation.
It's now automated now - we would have to think about it.
Could one of you download the newest cygwin.exe and try to update the whole system checking if everything works?
If yes, we could automate the process easily...

**Comment by tombujok** [comment]
I was not specific enough - I meant:
Download Cygwin installer from http://cygwin.com/setup-x86.exe
Run the installer and use your .babun\cygwin directory (most likely C:\Users\your username\.babun\cygwin) as the root install directory
Click through the installer. By default, it upgrades all packages.

**Comment by tombujok** [comment]
So one guy has already confirmed that updating with cygwin's setup.exe is the way to go:
https://twitter.com/virtualblackfox/status/563016897852354561

**Comment by p3lim** [comment]
Can confirm that worked, although it is important that the right mirror is selected for each user, as it can vary from a 2 minute update to a 2 hour update.

**Comment by v-yadli** [comment]
I can confirm that it works for me. Thanks! But now it takes a long time to spawn a shell. Don't know what happened. I'll try rebase everything manually.

**Comment by RMAdevelopper** [comment]
Hello,

Thank you for your help.
It worked with the manual cygwin update. (Haven't yet try to perform a rebase)

However Babun launch time has become very long.

**Comment by v-yadli** [comment]
@RMAdevelopper This worked for me:
http://stackoverflow.com/questions/28410852/startup-is-really-slow-for-all-cygwin-applications

Turns out that the slow start of mintty is due to slow credential access. I'm in a domain network and the lag was around 30 secs. After getting home (away from the domain network) cygwin became completely inaccessible.

The solution in the above link fixed my problem. Could you please try that?

**Comment by v-yadli** [comment]
@RMAdevelopper also if you migrated your babun dist from one machine from another, you might want to do `mkpasswd -l -c > /etc/passwd` and `mkgroup -l -c /etc/group` just like what @tombujok made for us in the post install script.

**Comment by RMAdevelopper** [comment]
Will try the solution about slow start as soon as I am back at work(in 15days btw..) and complete this thread to let anyone knows.
Thanks for the advice!

**Comment by sgregson** [comment]
Also succesful upgrade using the cygwin approach listed in https://github.com/babun/babun/issues/232#issuecomment-72896728

After the install (updated to git 2.1.4) my difftool command was broken due to the perl error module problems outlined here - http://permalink.gmane.org/gmane.os.cygwin/151848. If you run into similar errors, just re-run the cygwin installer and update the `perl-Error` module to the lastest.

**Comment by tombujok** [comment]
Fixed in 1.1.0. 
Release status: #266

**Comment by eyeloveu** [comment]
After I pact install the gcc-g++, the g++ command outputs nothing, and it doesn't have any response. 
Thanks to @tombujok 's sulotion. Manually update the Cygwin work for me.
